File: LAB6.cpp
#include <iostream>

using namespace std;

double function(double n) {
    if (n == 0)
        return 1;
    else
        return (n / (n + 2) - 10) * function(n - 1);
}

double function() {
    int number2;
    cout << "Enter the number: " << endl;
    cin >> number2;
    double product = 1;
    while (number2 != 0) {
        product = product * (((double) number2 / (double) (number2 + 2)) - 10);
        number2--;
    }
    cout << "Your result is: " << product << endl;
}

int main() {
    int number;
    cout << "Enter the number: " << endl;
    cin >> number;
    cout << "Your result is: " << function(number) << endl;
    function();
    return 0;
}

File: LAB6.py
calculation = lambda n: (n - 3) / n ** 2
number = int(input("Enter the number: "))
List = []
for i in range(1, number + 1, 1):
    List.append(calculation(i))
print("Your result is:", sum(List))


def calculation2(n):
    """This function calculates product recursively """
    if n == 0:
        return 1
    else:
        return (n / (n + 2) - 10) * calculation2(n - 1)


number2 = int(input("Enter the number: "))
print(calculation2.__doc__)
print(calculation2(number2))
